spec/lib/intersection_validator_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "spec_helper"
+require_relative "../shared.rb"
+
+RSpec.describe IntersectionValidator do
```

Comment:
I will refactor the specs after merge this pull request, because this classes is starting to get too complicated to understand.